Source/CombatExtended/CombatExtended/Verbs/Verb_LaunchProjectileCE.cs

```diff
@@ -489,6 +489,9 @@ public virtual void ShiftTarget(ShiftVecReport report, bool calculateMechanicalO
 
                     }
                     targetHeight = targetRange.Average;
+		    if (targetHeight > CollisionVertical.WallCollisionHeight && report.roofed ) {
+			targetHeight = CollisionVertical.WallCollisionHeight;

```

Comment:
Shouldn't targetHeight be set to the average of the minimum height at which the target is exposed and roof height?  Otherwise half your rounds will hit the roof instead of the target.

coverRange.max is the lowest you can shoot and still hit the target right?

Reply:
Actually, I suggest that it go:
```
if targetRange.max > roof && rooofed
   targetRange.max = roof
   if targetRange.min > roof
        targetRange.min = coverRange.max
targetHeight = targetRange.Average;
```